californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -397,35 +477,41 @@ public void sendResponse(final Exchange exchange, final Response response) {
 	@Override
 	public void receiveResponse(final Exchange exchange, final Response response) {
 
-		if (exchange.getRequest().isCanceled()) {
-			// do not continue fetching blocks if canceled
-			// reject (in particular for Block+Observe)
-			if (response.getType()!=Type.ACK) {
-				LOGGER.finer("rejecting blockwise transfer for canceled Exchange");
-				EmptyMessage rst = EmptyMessage.newRST(response);
-				sendEmptyMessage(exchange, rst);
-				// Matcher sets exchange as complete when RST is sent
-			}
+		if (isTransparentBlockwiseHandlingEnabled()) {
 
-		} else if (!response.hasBlockOption()) {
+			if (exchange.getRequest().isCanceled()) {
```

Comment:
I think, I begin to understand, that "non" is "none" (no blockwise response) and not NON (Non Confirmed).
So I agree, delete it from the blockwise layer.
There maybe something in the observer layer (NON/CON Notifies) and something in the ReliabilityLayer (CON).
But just sending RST to all none ACK seems to be strange to me.